### PR TITLE
Tell users about how VOLUME initializes the new mount point/volume

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
LGTM from a technical point of view.

**Comment by cpuguy83** [comment]
But we may want to add a note in that this does _not_ happen for bind-mounts or with volumes-from.

**Comment by duglin** [comment]
@cpuguy83 to make sure I grok it... you mean it doesn't apply when:
- you specify an existing host path on the -v
- attaching to another container's volume
right?

**Comment by cpuguy83** [comment]
@duglin Correct

**Comment by duglin** [comment]
@cpuguy83 ok will adjust accordingly - thanks

**Comment by duglin** [comment]
Updated

**Comment by estesp** [comment]
looks like a great addition to the docs on a potentially confusing area for the uninitiated.  @moxiegirl, want to do your review magic? :)

**Comment by duglin** [comment]
oh no!  now I'll have to change the entire section!  :-)

File: docs/sources/userguide/dockervolumes.md [hunk 1]
```diff
@@ -132,6 +134,11 @@ And another:
 
     $ sudo docker run -d --volumes-from dbdata --name db2 training/postgres
 
+It is important to note that if the `postgres` image contained a directory
+called `/dbdata` then mounting the volumes from the `dbdata` container will
+overlay (hide) the `/dbdata` files from the `postgres` image. Meaning, only
```
Comment (cpuguy83) [inline docs/sources/userguide/dockervolumes.md]: Might not use `overlay` here since it's also a supported fs type, which of course merges things.

Comment (duglin) [inline docs/sources/userguide/dockervolumes.md]: done!

File: docs/sources/userguide/dockervolumes.md [hunk 2]
```diff
@@ -24,7 +24,9 @@ containers that bypasses the [*Union File
 System*](/terms/layer/#union-file-system) to provide several useful features for
 persistent or shared data:
 
-- Volumes are initialized when a container is created
+- Volumes are initialized when a container is created. Any data in the image
```
Comment (moxiegirl) [inline docs/sources/userguide/dockervolumes.md]: The bullet is a bit awkward to parse ...maybe this
--------
## Data volumes

A *data volume* is a specially-designated directory within one or more
containers that bypasses the [*Union File
System*](/terms/layer/#union-file-system). Data volumes provide several useful features for persistent or shared data:

- Volumes are initialized when a container is created. If the container's base image contains data at the specified mount point, that data is copied into the new volume.
- Data volumes can be shared and reused among containers.
- Changes to a data volume are made directly.
- Changes to a data volume are not be included when you update an image.
- Data volumes persist even if the container itself is deleted.

Data volumes are designed to persist data, independent of the container's life 
cycle. Docker therefore *never* automatically deletes volumes when you remove 
a container, nor does it "garbage collect" volumes the container no longer references.

File: docs/sources/userguide/dockervolumes.md [hunk 3]
```diff
@@ -132,6 +134,11 @@ And another:
 
     $ sudo docker run -d --volumes-from dbdata --name db2 training/postgres
 
+It is important to note that if the `postgres` image contained a directory
```
Comment (moxiegirl) [inline docs/sources/userguide/dockervolumes.md]: In this case, if the `postgres` image contained a directory
called `/dbdata` then mounting the volumes from the `dbdata` container 
hide sthe `/dbdata` files from the `postgres` image. The result is only
the files from the `dbdata` container are visible.

File: docs/sources/reference/builder.md [hunk 4]
```diff
@@ -782,9 +782,21 @@ and mark it as holding externally mounted volumes from native host or other
 containers. The value can be a JSON array, `VOLUME ["/var/log/"]`, or a plain
 string with multiple arguments, such as `VOLUME /var/log` or `VOLUME /var/log
 /var/db`.  For more information/examples and mounting instructions via the
-Docker client, refer to [*Share Directories via Volumes*](/userguide/dockervolumes/#volume)
+Docker client, refer to
```
Comment (moxiegirl) [inline docs/sources/reference/builder.md]: "Readers use technical documents to perform tasks or gather information. For readers, these activities take place in the present. Therefore, the present tense is appropriate in most cases."   --->  which is why I'm editing this.  All that future tense with passive voice is resulting in what will be very difficult to parse sentence structures. :-D  Not your fault but I'm trying to correct as we go.

---

## VOLUME

    VOLUME ["/data"]

The `VOLUME` instruction creates a mount point with the specified name
and marks it as holding externally mounted volumes from native host or other
containers. The value can be a JSON array, `VOLUME ["/var/log/"]`, or a plain
string with multiple arguments, such as `VOLUME /var/log` or `VOLUME /var/log
/var/db`.  For more information/examples and mounting instructions via the
Docker client, refer to 
[*Share Directories via Volumes*](/userguide/dockervolumes/#volume)
documentation.

The `docker run` command initializes the newly created volume with any data that
exists at the specified location within the base image. For example, consider the
following Dockerfile snippet:

    FROM ubuntu
    RUN mkdir /myvol
    RUN echo "hello world" > /myvol/greating
    VOLUME /myvol

This Dockerfile results in an image that causes `docker run`, to
create a new mount point at `/myvol` and copy the  `greating` file 
into the newly created volume.

> **Note**:
> The list is parsed as a JSON array, which means that
> you must use double-quotes (") around words not single-quotes (').

File: docs/man/docker-run.1.md [hunk 5]
```diff
@@ -336,6 +336,10 @@ mount is not running.
 default, the volumes are mounted in the same mode (read write or read only) as
 the reference container.
 
+   If the location of the volume from the specified container overlaps with
```
Comment (moxiegirl) [inline docs/man/docker-run.1.md]: Source/target or source/destination are good adjectives to use -- this helps keep the two locations distinct and reinforce the relationship in the reader's mind. You could apply source/target to the container or to the volume just be consistent. In this instance I used container as it made sense
-----

**--volumes-from**=[]
   Mount volumes from the specified container(s)

Mounts already mounted volumes from a source container onto one or more additional, target containers. You must supply the source container's container-id. To share a volume, use the **--volumes-from** option when running
the target containers. You can share volumes even if the source container is not
running.

 By default, Docker mounts the volumes the same mode (read-write or read-only)
 as it the mount on the source container. Optionally, you change this  by
 suffixing the container-id with either the `:ro` or `:rw ` keyword.

If the location of the volume from the source container overlaps with
data residing on a target container, then the volume hide
that data on the target.

**Comment by moxiegirl** [comment]
Hey, hey, hey... @estesp  @duglin --- I read your comments after I commented.  Hmm, I don't think they would have stopped me...bwahahahahahahah!  She's mad with the commenting power.

File: docs/man/docker-run.1.md (hunk 5 above)
Comment (duglin) [inline docs/man/docker-run.1.md]: @moxiegirl ok - picked up your edits, with just minor tweaks. See how it looks now...

**Comment by duglin** [comment]
@moxiegirl oh wait, I missed some - hold on....

**Comment by duglin** [comment]
@moxiegirl ok picked 'em up. Again, with minor tweaks but mainly copy-n-paste.  see what you think.

**Comment by moxiegirl** [comment]
Thanks Doug for the PR and the extra work. Looks Good To Me oh, right, LGTM

**Pull request merged**